api/user_interaction_handler.go

```diff
@@ -45,3 +45,39 @@ func (s *PushToUsersHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	Write(w, http.StatusOK, `{"success": true}`)
 
 }
+
+type kickMsg struct {
+	Uids         []string
+	FrontendType string
+}
+
+// KickUserHandler handler
+type KickUserHandler struct {
+	App *App
+}
+
+// NewKickUserHandler creates a new kick user handler
+func NewKickUserHandler(a *App) *KickUserHandler {
+	m := &KickUserHandler{App: a}
+	return m
+}
+
+func (s *KickUserHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	var kick kickMsg
+	err := json.NewDecoder(r.Body).Decode(&kick)
+
+	if err != nil {
+		Write(w, http.StatusBadRequest, `{"success":false, "message":"failed to decode request body into kick struct"}`)
```

Comment:
You could include a reason field in the response, with the error string... this would be useful when debugging why decoding failed. 
```
fmt.Sprintf(`{"success":false, "message":"failed to decode request body into kick struct", "reason": "%s"}`, err.Error())
```

also, add some logs on error cases before returning (will also help debugging and troubleshooting issues)